echopype/convert/set_groups_ek80.py

```diff
@@ -1311,31 +1318,88 @@ def set_vendor(self) -> xr.Dataset:
             ds_cal = ds_cal.rename_vars({"impedance": "impedance_transducer"})
 
         #  Save decimation factors and filter coefficients
-        coeffs = dict()
-        decimation_factors = dict()
+        coeffs_and_decimation = {
+            t: {FILTER_IMAG: [], FILTER_REAL: [], DECIMATION: []}
+            for t in [WIDE_BAND_TRANS, PULSE_COMPRESS]
+        }
+
         for ch in self.sorted_channel["all"]:
             # filter coeffs and decimation factor for wide band transceiver (WBT)
             if self.parser_obj.fil_coeffs and (ch in self.parser_obj.fil_coeffs.keys()):
-                coeffs[f"{ch} WBT filter"] = self.parser_obj.fil_coeffs[ch][1]
-                decimation_factors[f"{ch} WBT decimation"] = self.parser_obj.fil_df[ch][1]
+                coeffs_vals = self.parser_obj.fil_coeffs[ch][1]
+                coeffs_and_decimation[WIDE_BAND_TRANS][FILTER_IMAG].append(np.imag(coeffs_vals))
+                coeffs_and_decimation[WIDE_BAND_TRANS][FILTER_REAL].append(np.real(coeffs_vals))
+                coeffs_and_decimation[WIDE_BAND_TRANS][DECIMATION].append(
+                    self.parser_obj.fil_df[ch][1]
+                )
             # filter coeffs and decimation factor for pulse compression (PC)
             if self.parser_obj.fil_df and (ch in self.parser_obj.fil_coeffs.keys()):
-                coeffs[f"{ch} PC filter"] = self.parser_obj.fil_coeffs[ch][2]
-                decimation_factors[f"{ch} PC decimation"] = self.parser_obj.fil_df[ch][2]
+                coeffs_vals = self.parser_obj.fil_coeffs[ch][2]
+                coeffs_and_decimation[PULSE_COMPRESS][FILTER_IMAG].append(np.imag(coeffs_vals))
+                coeffs_and_decimation[PULSE_COMPRESS][FILTER_REAL].append(np.real(coeffs_vals))
+                coeffs_and_decimation[PULSE_COMPRESS][DECIMATION].append(
+                    self.parser_obj.fil_df[ch][2]
+                )
 
         # Assemble everything into a Dataset
         ds = xr.merge([ds_table, ds_cal])
 
-        # Save filter coefficients as real and imaginary parts as attributes
-        for k, v in coeffs.items():
-            ds.attrs[k + "_r"] = np.real(v)
-            ds.attrs[k + "_i"] = np.imag(v)
-
-        # Save decimation factors as attributes
-        for k, v in decimation_factors.items():
-            ds.attrs[k] = v
+        # Add the coeffs and decimation
+        ds = ds.pipe(self._add_cd, coeffs_and_decimation)
 
         # Save the entire config XML in vendor group in case of info loss
         ds.attrs["config_xml"] = self.parser_obj.config_datagram["xml"]
 
         return ds
+
+    @staticmethod
+    def _add_cd(
```

Comment:
```suggestion
    def _add_filter_params(
```
Trying to make it more readable for our future selves...